appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -225,6 +247,51 @@ public void onCreate(Bundle icicle) {
     viewLayout = new LinearLayout(this, ComponentConstants.LAYOUT_ORIENTATION_VERTICAL);
     alignmentSetter = new AlignmentUtil(viewLayout);
 
+    progress = null;
+    if (!_initialized && formName.equals("Screen1")) {
```

Comment:
Are we setting ourselves up to lose in the future, if we try to generalize things so that the initial screen doesn't need to be named Screen1?   Can we add a warning comment in the code somewhere ?